src/System.Security.Cryptography.Xml/tests/TestHelpers.cs

```diff
@@ -2,46 +2,120 @@
+        /// The <see cref="Stream"/> to read from. This cannot be null.
+        /// </param>
+        /// <param name="encoding">
+        /// The <see cref="Encoding"/> to use. This cannot be null.
+        /// </param>
+        /// <returns>
+        /// The stream as a string.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        /// No argument can be null.
+        /// </exception>
+        public static string StreamToString(Stream stream, Encoding encoding)
         {
-            if (testName == null)
-                throw new ArgumentNullException(nameof(testName));
+            if (stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
+            }
+            if (encoding == null)
+            {
+                throw new ArgumentNullException(nameof(encoding));
+            }
 
-            var file = new TempFile(
-                Path.Combine(Directory.GetCurrentDirectory(), testName + ".dtd")
-            );
-
-            File.WriteAllText(file.Path, "<!-- presence, not content, required -->");
-
-            return file;
+            using (StreamReader streamReader = new StreamReader(stream, encoding))
+            {
+                return streamReader.ReadToEnd();
+            }
         }
 
-        public static TempFile CreateTestTextFile(string testName, string content)
+        /// <summary>
+        /// Perform
+        /// </summary>
+        /// <param name="inputXml">
+        /// The XML to transform. This cannot be null, empty or whitespace.
+        /// </param>
+        /// <param name="transform">
+        /// The <see cref="Transform"/> to perform on 
+        /// <paramref name="inputXml"/>. This cannot be null.
+        /// </param>
+        /// <param name="encoding">
+        /// An optional <see cref="Encoding"/> to use when serializing or 
+        /// deserializing <paramref name="inputXml"/>. This should match the 
+        /// encoding specified in <paramref name="inputXml"/>. If omitted or 
+        /// null, <see cref="UTF8Encoding"/> is used.
+        /// </param>
+        /// <param name="resolver">
+        /// An optional <see cref="XmlResolver"/> to use. If omitted or null, 
+        /// no resolver is used.
+        /// </param>
+        /// <returns>
+        /// The transformed <paramref name="inputXml"/>.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="transform"/> cannot be null.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        /// <paramref name="inputXml"/> cannot be null, empty or whitespace.
+        /// </exception>
+        /// <exception cref="XmlException">
+        /// <paramref name="inputXml"/> is not valid XML.
+        /// </exception>
+        public static string ExecuteTransform(string inputXml, Transform transform, Encoding encoding = null, XmlResolver resolver = null)
```

Comment:
Since this code is probably going to be shared across multiple test classes, I wanted to document it properly, including exceptions, so Intellisense will give the correct values.